### PR TITLE
feat: Operational metrics for metronome integration

### PR DESCRIPTION
**Pull request by adilansari**
## Describe your changes
- Capture and report latency and http responses from Metronome

## How best to test these changes
- unit tests

## Issue ticket number and link
TIG-1239

### CODE REVIEW

File: server/services/v1/billing/metronome.go [hunk 1]
```diff
@@ -50,18 +52,52 @@ func NewMetronomeProvider(config config.Metronome) (*Metronome, error) {
...
+
+	resp, err := f(ctx)
+	me.RecordDuration(metrics.MetronomeLatency, me.GetTags())
+	if resp != nil {
+		defer resp.Body.Close()
+		me.IncrementCount(metrics.MetronomeResponseCode, me.GetTags(), strconv.Itoa(resp.StatusCode), 1)
+	}
+
+	var errCount int64
+	if err != nil {
```
Comment (himank) [inline server/services/v1/billing/metronome.go:70]: Should we log it as well?

Comment (adilansari) [reply server/services/v1/billing/metronome.go:70]: So its getting returned up the stack. We should be able to see a error log from upstream method call.

Comment (himank) [reply server/services/v1/billing/metronome.go:70]: sg

**Review by himank — DISMISSED** [review]

**Review by himank — APPROVED** [review]

File: server/services/v1/billing/reporter.go [hunk 2]
```diff
@@ -95,6 +95,8 @@ func (r *UsageReporter) push() error {
 			log.Error().Msgf("invalid namespace id %s", namespaceId)
 			continue
 		}
+		// todo: remove this extraneous logging
+		log.Info().Msgf("received usage for id %s - %d, %d, %d", namespaceId, stats.ReadUnits, stats.WriteUnits, stats.SearchUnits)
```
Comment (pboros) [inline server/services/v1/billing/reporter.go:99]: Should this be debug?

**Pull request merged**